x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/filter/SecurityActionFilter.java

```diff
@@ -109,6 +114,16 @@ operations are blocked on license expiration. All data operations (read and writ
                     TransportVersion.current(), // current version since this is on the same node
                     (original) -> { applyInternal(task, chain, action, request, contextPreservingListener); }
                 );
+            } else if (secondaryAuthActions.get().contains(action)) {
+                SecondaryAuthentication secondaryAuth = securityContext.getSecondaryAuthentication();
+                if (secondaryAuth == null) {
+                    throw new IllegalArgumentException("es-secondary-authorization header must be used to call action [" + action + "]");
+                } else {
+                    secondaryAuth.execute(ignore -> {
+                        applyInternal(task, chain, action, request, contextPreservingListener);
+                        return null;
+                    });
```

Comment:
I think there's a problem here that this eats up the `_xpack_security_secondary_authc`.
A few actions rely on it to be present, see for eg `org.elasticsearch.xpack.transform.utils.SecondaryAuthorizationUtils`.

But, more importantly, the action filter is executed across the nodes.
I was able to make `SecondaryAuthActionIT` fail for the _get index_ action, which forwards the request to the master node, with:
```
diff --git a/x-pack/plugin/security/qa/secondary-auth-actions/src/javaRestTest/java/org/elasticsearch/secondary/auth/actions/SecondaryAuthActionsIT.java b/x-pack/plugin/security/qa/secondary-auth-actions/src/javaRestTest/java/org/elasticsearch/secondary/auth/actions/SecondaryAuthActionsIT.java
index 8f593d51bb1..e793912adfb 100644
--- a/x-pack/plugin/security/qa/secondary-auth-actions/src/javaRestTest/java/org/elasticsearch/secondary/auth/actions/SecondaryAuthActionsIT.java
+++ b/x-pack/plugin/security/qa/secondary-auth-actions/src/javaRestTest/java/org/elasticsearch/secondary/auth/actions/SecondaryAuthActionsIT.java
@@ -37,6 +37,7 @@ public class SecondaryAuthActionsIT extends ESRestTestCase {
     @ClassRule
     public static ElasticsearchCluster cluster = ElasticsearchCluster.local()
         .distribution(DistributionType.DEFAULT)
+        .nodes(2)
         .setting("xpack.watcher.enabled", "false")
         .setting("xpack.ml.enabled", "false")
         .setting("xpack.security.enabled", "true")
```

Reply:
thanks Albert !  

I fixed this in https://github.com/elastic/elasticsearch/pull/106613/commits/f3a4957e43be8dee6f51aa7103754e9f1d54406e

Reply:
>A few actions rely on it to be present, see for eg org.elasticsearch.xpack.transform.utils.SecondaryAuthorizationUtils.

The actions names used here won't overlap with the pre-existing actions that use Secondary Auth already.